ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,214 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
+import { I18nBehavior } from '../nuxeo-i18n-behavior.js';
+import '@nuxeo/nuxeo-elements/nuxeo-element.js';
+import '../../core/nuxeo-page-provider.js';
+import '../search/nuxeo-search-form-layout.js';
+import '../search/nuxeo-search-results-layout.js';
+import '../widgets/nuxeo-dialog.js';
+import '@polymer/paper-button/paper-button.js';
+
+{
+  /**
+   * `nuxeo-document-picker`
+   * @group Nuxeo UI
+   * @element nuxeo-document-picker
+   */
+  class DocumentPicker extends mixinBehaviors([I18nBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style include="nuxeo-styles">
+          nuxeo-dialog {
+            display: flex;
+            flex-direction: column;
+            justify-content: space-between;
+            height: 100%;
+            width: 100%;
+            /* TODO check if we can use something a bit bigger here */
+            max-height: 84%;
```

Comment:
shouldn't this be set at application level instead of hardcoded in the element itself ?

Reply:
Where would we set such values in Web UI? [`base.js`](https://github.com/nuxeo/nuxeo-web-ui/blob/master/themes/base.js)?
Would we leave them unset at the element level?
What's supposed to happen if the person using the element doesn't set them?

Reply:
@nelsonsilva, you mean exposing some sort of css variable?

Reply:
I think we would have to expose a CSS variable for this because limiting the size of the `<nuxeo-document-picker>` (from the outside) doesn't limit the size of the `<nuxeo-dialog>` in `<nuxeo-document-picker>`.

Reply:
@mnixo: Even if we set the dialog's height to 100%?

Reply:
This is the result of moving `max-height` and `max-width` to `<nuxeo-html-editor>`. The dialog `height` and `width` on `<nuxeo-document-picker>` are still set to `100%`. The dialog ends up taking all the available space.

https://user-images.githubusercontent.com/4223240/112303821-a79c0300-8c94-11eb-8b6f-08158f7abf7d.mp4

Reply:
Indeed. I think this happens because the positioning of the dialog is fixed, which means the [containing block will be the window](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block). And I think this behavior is introduce by [IronFitBehavior](https://github.com/PolymerElements/iron-fit-behavior/blob/master/iron-fit-behavior.js#L420). I think the variable is our only alternative here.